speculation-rules.bs

```diff
@@ -300,6 +427,40 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
   We should also cancel speculated prerenders.
 </p>
 
+<h3 id="document-rule-predicate-matching">Document rule predicate matching</h3>
+
+<div algorithm>
+  A [=document rule predicate=] |predicate| <dfn for="document rule predicate">matches</dfn> an {{Element}} |el| implementing the {{HTMLHyperlinkElementUtils}} mixin if the following steps return true:
+
+  1. If |predicate| is a [=document rule conjunction=], then:
+    1. [=list/For each=] |clause| of |predicate|'s [=document rule conjunction/clauses=]:
+      1. If |clause| does not [=document rule predicate/match=] |el|, return false.
+    1. Return true.
+  1. If |predicate| is a [=document rule disjunction=], then:
+    1. [=list/For each=] |clause| of |predicate|'s [=document rule disjunction/clauses=]:
+      1. If |clause| [=document rule predicate/matches=] |el|, return true.
+    1. Return false.
+  1. If |predicate| is a [=document rule negation=], then:
+    1. If |predicate|'s [=document rule negation/clause=] [=document rule predicate/matches=] |el|, return false.
+    1. Return true.
+  1. If |predicate| is a [=document rule URL pattern predicate=], then:
+    1. Let |href| be the result of running |el|'s {{HTMLHyperlinkElementUtils/href}} getter steps.
+    1. [=list/For each=] |pattern| of |predicate|'s [=document rule URL pattern predicate/patterns=]:
+      1. Execute |pattern|'s {{URLPattern/test(input, baseURL)}} steps given |href|. If the reuslt is true, return true.
```

Comment:
Let's use the https://wicg.github.io/urlpattern/#match operation directly

Also, typo "reuslt"

Reply:
Done.